### PR TITLE
chore: add redos-detector GitHub Action

### PR DESCRIPTION
**Pull request by Trott**

### CODE REVIEW

**Comment by Trott** [comment]
Putting in Draft mode until 1.0.1 of the action workflow is released.

**Comment by meekdenzo** [comment]
v1.0.1 is live now https://github.com/marketplace/actions/vuln-regex-detector

**Comment by Trott** [comment]
I intentionally added a ReDoS in https://github.com/Trott/slug/pull/272. Hopefully the GitHub Action detects it!

**Comment by Trott** [comment]
Also, the GitHub Action ran in less than 10 minutes, which is faster than I was expecting it to take with all the installation and setup it has to do.

**Comment by Trott** [comment]
The good news is the job tries to scan the changed files. The bad news is it can't find the Perl script it needs to do the scanning.

```
Scanning for vulnerable regexes in slug.js
Can't open perl script "./bin/check-file.pl": No such file or directory
The following vulnerable regexes were found in slug.js
```

File: .github/workflows/redos-detector.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with: 
+          fetch-depth: '0'
+ 
+      - name: Scan for redos
+        uses: meekdenzo/vrd@v1.0.2
```
Comment (Trott) [inline .github/workflows/redos-detector.yml:17]:
```suggestion
        uses: meekdenzo/vrd@v1.0.3
```

**Pull request merged**